php: 8.3 Support

JediKev opened the pull request:
This includes the needed changes to fully support PHP 8.3. This also updates the PHP versions in the Requirements section in the README to reflect the latest supported versions.

JediKev committed:
deprecation: get_class() No Parameters

In PHP 8.3 they have deprecated the use of `get_class()` with no parameters.
Instead you should use `get_class($this)`, `self::class`, or `__CLASS__`.
This updates all calls in the code to make it 8.3 compliant. These changes
are backwards compatible with PHP 8.1-8.2.

JediKev committed:
update: README PHP Version

This updates the PHP version in the Requirements section from `8.0 - 8.2
(8.2 recommended)` to `8.1 - 8.3 (8.3 recommended)`.

JediKev commented:
We still need to review the [`assert_options()` deprecation](https://php.watch/versions/8.3/assert-multiple-deprecations) (with no replacement) [we use in our tests](https://github.com/search?q=repo%3AosTicket%2FosTicket%20assert_options&type=code).

JediKev committed:
issue: Cannot Redeclare mb_str_pad()

This addresses an issue where in PHP 8.3 our declaration of `mb_str_pad()`
conflicts with the native `mb_str_pad()`. This adds a check to see if the
function already exists or not before declaring it.

protich commented:
> We still need to review the [`assert_options()` deprecation](https://php.watch/versions/8.3/assert-multiple-deprecations) (with no replacement) [we use in our tests](https://github.com/search?q=repo%3AosTicket%2FosTicket%20assert_options&type=code).

I'm not sure we even need the assertion - the class `Test` defines the routine.

protich reviewed (approved):
Looks good... before merging we need confirmation that core plugins are also compatiple with PHP 8.3.

JediKev committed:
update: System Prerequisites (PHP)

This updates the PHP version in the System Prerequisites (used in the
installer and upgrader) from 8.0 to 8.1. Since upstream PHP dropped support
for it we will too. RIP

JediKev committed:
deprecated: assert_options()

This removes a call to a deprecated function `assert_options()` to match
PHP 8.3 changes.
